my_enumerable.rb

```diff
@@ -0,0 +1,164 @@
+# rubocop: disable Metrics/PerceivedComplexity: Perceived complexity for my_any? is too high. [11/7]
+# rubocop: disable Metrics/CyclomaticComplexity: Cyclomatic complexity for my_any? is too high. [10/6]
+# rubocop: disable Metrics/ModuleLength: Module has too many lines. [135/100]
+# rubocop: disable Style/For: Prefer each over for
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    item = 0
+    array = to_a
+    n = array.length
+    while item < n
+      yield(array[item])
+      item += 1
+    end
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    i = 0
+    for item in self do
+      yield(item, i)
+      i += 1
+    end
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    new_arr = []
+    for item in self do
+      new_arr.push(item) if yield(item)
+    end
+    new_arr
+  end
+
+  def my_all?(*args)
+    if args.nil?
+      my_each do |item|
+        return false unless args == item
+      end
+    elsif block_given?
+      my_each do |item|
+        return false unless yield(item)
+      end
+    else
+      my_each do |item|
+        return false unless item
+      end
+    end
+    true
+  end
+
+  def my_any?(args = nil)
+    if !block_given? && args.nil?
+      my_each do |item|
+        return true if item
+      end
+    elsif args.is_a? Regexp
+      my_each do |item|
+        return true if args == item
+      end
+    elsif args.is_a? Array
+      my_each do |item|
+        return true if yield(item)
+      end
+    elsif args.is_a? Hash
+      my_each do |x, y|
+        return true if yield(x, y)
+      end
+    end
+    false
+  end
+
+  def my_none?(args = nil)
+    if !block_given? && args.nil?
+      my_each do |item|
+        return false if item
+      end
+    elsif args.is_a? Regexp
+      my_each do |item|
+        return false if item == args
+      end
+    elsif args.is_a? Array
+      my_each do |item|
+        return false if yield(item)
+      end
+    elsif args.is_a? Hash
+      my_each do |y, z|
+        return false if yield(y, z)
+      end
+    end
+    true
+  end
```

Comment:
`my_none?` identical and returns the same thing as ruby's [`none?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-none-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_none.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
- returns true if the block never returns true for all elements
```ruby
[1, 2, 3].my_none? { |num| num > 4 }  #should return true.
```
- when a class is passed as an argument returns true if none of the collection is a member of such class
```ruby
[1, 2, 3].my_none?(String)  #should return true
['1', '2', '3'].my_none?(String)  #should return false
```
- when a Regex is passed as an argument returns true only if none of the collection matches the Regex
```ruby
%w[dog door rod blade].my_none?(/z/)   #should return true
%w[dog door rod blade].my_none?(/d/)   #should return false
```
- when a pattern other than Regex or a Class is given returns true only if none of the collection matches the pattern
```ruby
[1, 2, 4].my_none?(3) #should return true
[1, 2, 3].my_none?(3) #should return false
```